### PR TITLE
Reimplementation of Google analytics with cookie consent preferences management

### PR DESCRIPTION
**Pull request by kr8n3r**
What
----

A reimplementation of previous tracking - #546 - but only once users consent to it.

It allows users to opt-in or opt-out and not see the banner again for a year

How to review
-------------

- check out branch
- run npm install
- set `this.cookieDomain` to empty string for localhost test in cookie-functions.js
- start the app
- visit http://localhost:3000
- if you haven't set a preference you will see a banner
- save a cookie preference
- check if have cookies stored (chrome: Option + ⌘ + J (on macOS), or Shift + CTRL + J (on Windows/Linux) - Application tab

Who can review
---------------

not @kr8n3r

### CODE REVIEW

File: src/frontend/javascript/cookie-functions.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+Cookies.prototype.initCookieBanner = function ($module) {
+  this.$module = $module
+
+  this.$module.hideCookieMessage = this.hideCookieMessage.bind(this)
+  this.$module.showBannerConfirmationMessage = this.showBannerConfirmationMessage.bind(this)
+  this.$module.setBannerCookieConsent = this.setBannerCookieConsent.bind(this)
+  this.$module.cookieBannerConfirmationMessage = this.$module.querySelector('.cookie-banner__confirmation')
+  
+  /* test already test this function */
```
Comment (46bit) [inline src/frontend/javascript/cookie-functions.js:38]: what does this comment mean?

File: src/frontend/javascript/cookie-functions.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+  if (!this.$module) {
```
Comment (46bit) [inline src/frontend/javascript/cookie-functions.js:40]: What is this check trying to achieve? It seems out of place given we already set a bunch of properties on the lines above.

File: src/frontend/javascript/cookie-functions.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  }
+
+  this.showCookieBanner()
+}
+
+Cookies.prototype.showCookieBanner = function () {
+  // Show the cookie banner if there is no cookie set
+  var hasCookiesPolicy = this.getCookie(this.cookieName)
+  if (this.$module && !hasCookiesPolicy) {
+  this.$module.style.display = 'block'
```
Comment (46bit) [inline src/frontend/javascript/cookie-functions.js:66]:
```suggestion
    this.$module.style.display = 'block'
```

File: src/frontend/javascript/cookie-functions.js (hunk 2 above)
Comment (kr8n3r) [reply src/frontend/javascript/cookie-functions.js:40]: if there is no dom element don't run the rest

File: src/frontend/javascript/cookie-functions.js (hunk 1 above)
Comment (kr8n3r) [reply src/frontend/javascript/cookie-functions.js:38]: forgot to remove, after i changed the tests
 will update

**Review by 46bit — APPROVED** [review]
This works really well :)

**Pull request merged**